tests/manage/z_cluster/nodes/test_disk_failures.py

```diff
@@ -123,16 +149,28 @@ def test_detach_attach_2_data_volumes(self, nodes, pvc_factory, pod_factory):
 
         for worker_and_volume in workers_and_volumes:
             # Detach the volume (logging is done inside the function)
-            nodes.detach_volume(
-                worker_and_volume['volume'], nodes.detach_volume(worker_and_volume['worker'])
-            )
-
-        for worker_and_volume in workers_and_volumes:
-            # Wait for worker volume to be re-attached automatically to the node
-            assert nodes.wait_for_volume_attach(worker_and_volume['volume']), (
-                f"Volume {worker_and_volume['volume']} "
-                f"failed to be re-attached to a worker node"
-            )
+            try:
+                nodes.detach_volume(
+                    worker_and_volume['volume'], worker_and_volume['worker']
+                )
+            except AWSTimeoutException as e:
+                if "Volume state: in-use" in e:
+                    logger.info(
+                        f"Volume {worker_and_volume['volume']} re-attached "
+                        f"successfully to worker "
+                        f"node {worker_and_volume['worker']}")
+                else:
+                    raise
+            else:
+                """
+                Wait for worker volume to be re-attached automatically
+                to the node
+                """
+                assert nodes.wait_for_volume_attach(
+                    worker_and_volume['volume']
+                ), (f"Volume {worker_and_volume['volume']} failed to be "
+                    f"re-attached to a worker node"
+                    )
```

Comment:
pretty much similar to lines 92-110 (https://github.com/red-hat-storage/ocs-ci/pull/2214/files#diff-d67bf88a5b02bb04cb2c3834aa727b19R92-R110). 
Please create a class method and call it from both places.

Reply:
Done